### ISSUE TITLE
get_filename_component doesn't cover .t.cpp

### ISSUE THREAD

**Issue by giacomini** [RWTH-HPC/CMake-codecov#18]
Some source code layout guidelines suggest to name test source files with the combined extension `.t.cpp`. Unfortunately the use of `get_filename_component` at https://github.com/RWTH-HPC/CMake-codecov/blob/9a24e83a901e4acee01786ff2c75559f580bb772/cmake/Findcodecov.cmake#L136 is not adequate in such a situation, because it extracts the maximal extension `.t.cpp` instead of just `.cpp`, which makes the source file not recognizable.

An alternative would be the use of a regular expression, i.e. instead of:
```
get_filename_component(FILE_EXT "${FILE}" EXT)
string(TOLOWER "${FILE_EXT}" FILE_EXT)
string(SUBSTRING "${FILE_EXT}" 1 -1 FILE_EXT)
```
one could use
```
string(REGEX REPLACE "^.*\\.([^\\.]+)$" "\\1" FILE_EXT "${FILE}")
string(TOLOWER "${FILE_EXT}" FILE_EXT)
```

**Comment by Twon** [comment]
I've also just run into this issue.  Actually get_filename_component() support the parameter LAST_EXT which will then correctly return just the ".cpp" component of the filename (no need to a regex): https://cmake.org/cmake/help/latest/command/get_filename_component.html

This is just a one-line fix so I'll submit a pull request for this shortly.

**Comment by Twon** [comment]
Pull request submitted with a fix here: https://github.com/RWTH-HPC/CMake-codecov/pull/19

**Comment by alehaa** [comment]
Merged #19.

**Comment by giacomini** [comment]
Be careful, though, that LAST_EXT is available only in recent versions of CMake (3.14, I think).

**Comment by alehaa** [comment]
This has been fixed in ea06ae7c840195e30563b0c71b8a11766b338a14.

**Comment by giacomini** [comment]
Ok, fine with me, I can upgrade.
